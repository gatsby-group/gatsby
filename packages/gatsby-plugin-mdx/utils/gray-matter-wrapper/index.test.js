const grayMatterWrapper = require(`.`)
const c = require(`js-combinatorics`)

const generateExcerptFixture = params => {
  const { frontmatter, excerpt, excerpt_separator } = params
  const code = {
    frontmatter: `---
one: two
excerpt: This is excerpt from frontmatter.
---`,
    excerpt: `This is an excerpt.
This is also an excerpt.`,
    excerptSeparator: `<!--more-->`,
    body: `This could be an excerpt.
---
This is content.`,
  }
  return {
    name:
      Object.entries(params)
        .filter(([_, v]) => !!v)
        .map(([k, _]) => k)
        .join(`-`) || `body`,
    input: [
      frontmatter ? code.frontmatter : ``,
      code.excerpt,
      excerpt_separator ? code.excerptSeparator : ``,
      code.body,
    ].join(`\n`),
    options: {
      grayMatter: {
        excerpt: excerpt,
        excerpt_separator: excerpt_separator
          ? code.excerptSeparator
          : undefined,
      },
    },
  }
}

const excerptFixtures = c
  .baseN([true, false], 3)
  .toArray()
  .map(([frontmatter, excerpt, excerpt_separator]) => {
    const { name, input, options } = generateExcerptFixture({
      frontmatter,
      excerpt,
      excerpt_separator,
    })
    return [name, input, options]
  })

describe(`gray-matter`, () => {
  describe(`excerpt`, () => {
    test.each(excerptFixtures)(`snapshot with %s`, (name, input, options) => {
      const matter = grayMatterWrapper(input, options)
      expect(matter).toMatchSnapshot()
    })
  })
})
