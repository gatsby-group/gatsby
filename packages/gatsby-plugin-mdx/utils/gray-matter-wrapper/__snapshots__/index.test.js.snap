// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gray-matter excerpt snapshot with body 1`] = `
Object {
  "content": "
This is an excerpt.
This is also an excerpt.

This could be an excerpt.
---
This is content.",
  "data": Object {},
}
`;

exports[`gray-matter excerpt snapshot with excerpt 1`] = `
Object {
  "content": "
This is an excerpt.
This is also an excerpt.

This could be an excerpt.
---
This is content.",
  "data": Object {
    "excerpt": "
This is an excerpt.
This is also an excerpt.

This could be an excerpt.
",
  },
}
`;

exports[`gray-matter excerpt snapshot with excerpt_separator 1`] = `
Object {
  "content": "
This is an excerpt.
This is also an excerpt.
<!--more-->
This could be an excerpt.
---
This is content.",
  "data": Object {
    "excerpt": "
This is an excerpt.
This is also an excerpt.
",
  },
}
`;

exports[`gray-matter excerpt snapshot with excerpt-excerpt_separator 1`] = `
Object {
  "content": "
This is an excerpt.
This is also an excerpt.
<!--more-->
This could be an excerpt.
---
This is content.",
  "data": Object {
    "excerpt": "
This is an excerpt.
This is also an excerpt.
",
  },
}
`;

exports[`gray-matter excerpt snapshot with frontmatter 1`] = `
Object {
  "content": "This is an excerpt.
This is also an excerpt.

This could be an excerpt.
---
This is content.",
  "data": Object {
    "excerpt": "This is excerpt from frontmatter.",
    "one": "two",
  },
}
`;

exports[`gray-matter excerpt snapshot with frontmatter-excerpt 1`] = `
Object {
  "content": "This is an excerpt.
This is also an excerpt.

This could be an excerpt.
---
This is content.",
  "data": Object {
    "excerpt": "This is excerpt from frontmatter.",
    "one": "two",
  },
}
`;

exports[`gray-matter excerpt snapshot with frontmatter-excerpt_separator 1`] = `
Object {
  "content": "This is an excerpt.
This is also an excerpt.
<!--more-->
This could be an excerpt.
---
This is content.",
  "data": Object {
    "excerpt": "This is excerpt from frontmatter.",
    "one": "two",
  },
}
`;

exports[`gray-matter excerpt snapshot with frontmatter-excerpt-excerpt_separator 1`] = `
Object {
  "content": "This is an excerpt.
This is also an excerpt.
<!--more-->
This could be an excerpt.
---
This is content.",
  "data": Object {
    "excerpt": "This is excerpt from frontmatter.",
    "one": "two",
  },
}
`;
